File: 01/main.pl
open my $fh, "<", "input.txt";
while(<$fh>) {
    my $floor = 0;
    my $bii = 0;
    my @chars = split "", $_;
    while (my ($i, $value) = each @chars) {
        $floor++ if $value eq "(";
        $floor-- if $value eq ")";
        if ($floor < 0) { $bii = $i + 1 unless($bii); }
    }
    print "Part1: $floor\nPart2: $bii\n";
}

File: 01/main.rs
use std::fs::File;
use std::io::Read;

fn main() {
    let mut content = String::new();
    File::open("input.txt")
        .expect("Error opening file")
        .read_to_string(&mut content)
        .expect("Error reading file");

    let mut floors = 0;
    let mut bii = 0;
    for (i, c) in content.chars().enumerate() {
        match c {
            '(' => floors += 1,
            ')' => floors -= 1,
            _ => {}
        }
        if floors < 0 {
            if bii == 0 {
                bii = i + 1
            }
        }
    }
    println!("Part1: {}", floors);
    println!("Part2: {}", bii);
}